File: backend/src/main/java/com/wedspot/backend/services/IBookingService.java
package com.wedspot.backend.services;

import java.util.List;

import com.wedspot.backend.Model.APIResponse;
import com.wedspot.backend.Model.BookingDTO;

public interface IBookingService {
    APIResponse<List<BookingDTO>> getAllBookings();

    APIResponse<List<BookingDTO>> getClientBookings(Long clientId);

    APIResponse<List<BookingDTO>> getVendorBookings(Long vendorId);

    APIResponse<BookingDTO> createBooking(com.wedspot.backend.Model.BookingRequest request);
}

File: backend/Services/IBookingService.cs
using backend.Models.Entities;

namespace backend.Services
{
    public interface IBookingService
    {
        Task<List<Booking>> GetAllBookings();
        Task<List<Booking>> GetBookingsByClient(long clientId);
        Task<List<Booking>> GetBookingsByVendor(long vendorId);
        Task<Booking> CreateBooking(Booking booking);
        Task<Booking?> UpdateBookingStatus(long id, string status);
    }
}
